internals-js/src/specs/joinSpec.ts

```diff
@@ -126,6 +133,25 @@ export class JoinSpecDefinition extends FeatureDefinition {
       joinEnumValue.addArgument('graph', new NonNullType(graphEnum));
     }
 
+    const joinDirective = this.addDirective(schema, 'directive').addLocations(
+      // Allow @join__directive at any location where @joinType or @joinField
+      // can appear, as well as on the schema element.
+      ...new Set([
+        DirectiveLocation.SCHEMA,
+        ...joinType.locations,
```

Comment:
Do we really want to support it on all these locations? The locations for source are much more narrow.

Reply:
https://github.com/apollographql/federation/pull/2894/commits/8127b6983975e49dd090f3df2bb7243c2d85de23